iXBRLViewerPlugin/viewer/src/js/search.js

```diff
@@ -99,21 +99,62 @@ export class ReportSearch {
         doneCallback();
     }
 
+    visibilityFilter(s, item) {
+        return item.isHidden() ? s.showHiddenFacts : s.showVisibleFacts;
+    }
+
+    periodFilter(s, item) {
```

Comment:
This is fine, but since you refactored something like this might be more generic:

```python
class ValueMatcher:
    name: str
    value: str
    func: Callable

class Filter:
    def __init__(name, value_matchers):
        self.name = name
        self.value_matchers = {
            v.value: v for v in value_matchers
        }

    def html():
         return f'<div class="control-label" data-i18n="inspector.{self.name.lower()}">' +
         self.name +
          '</div>' +
          f'<select id="search-filter-{self.name}">' +
          '\n'.join([f'<option value="{v.value}" data-i18n="inspector.{v.value}">{v.name}</option>' for v in self.value_matchers]) +
          '</select>'
         
    def applies(filterValue, item):
        return filters[value].func(item)
...
filters = [Filter('Concept Type', [
     ValueMatcher('ALL', '*': lambda item: return True),
     ValueMatcher('NUMERIC', 'numeric': lambda item: item.isNumeric()),
     ValueMatcher('TEXT', 'text': lambda item: not item.isNumeric())
])]
```

Reply:
This works well for dropdown filters, but not the visible/hidden checkboxes, and I anticipate some soon-to-be-implemented filters that might be more complex, like multi-select. I think once we have all those filters implemented we will be in a better place to revisit a larger refactor like this that takes over HTML generation.

Reply:
Yeah works for me